File: examples/src/Slides/Layout/AddLayoutSlides.cs
using Aspose.slides.Export

/*
This project uses Automatic Package Restore feature of NuGet to resolve Aspose.Slides for .NET API reference 
when the project is build. Please check https:#docs.nuget.org/consume/nuget-faq for more information. 
If you do not wish to use NuGet, you can manually download Aspose.Slides for .NET API from http:#www.aspose.com/downloads, 
install it and then add its reference to this project. For any issues, questions or suggestions 
please feel free to contact us using http:#www.aspose.com/community/forums/default.aspx
*/

namespace Aspose.slides.Examples.CSharp.slides.Layout
{
    class AddLayoutSlides
    {
        public static void Run()
        {
            #ExStart:AddLayoutSlides
            # The path to the documents directory.
            dataDir = RunExamples.GetDataDir_Slides_Presentations_Layout()

            # Instantiate Presentation class that represents the presentation file
            using (Presentation presentation = new Presentation(dataDir + "AccessSlides.pptx"))
            {
                # Try to search by layout slide type
                IMasterLayoutSlideCollection layoutSlides = presentation.Masters[0].LayoutSlides
                ILayoutSlide layoutSlide = layoutSlides.GetByType(SlideLayoutType.TitleAndObject) ?? layoutSlides.GetByType(SlideLayoutType.Title)

                if (layoutSlide == None)
                {
                    # The situation when a presentation doesn't contain some type of layouts.
                    # presentation File only contains Blank and Custom layout types.
                    # But layout slides with Custom types has different slide names,
                    # like "Title", "Title and Content", etc. And it is possible to use these
                    # names for layout slide selection.
                    # Also it is possible to use the set of placeholder shape types. For example,
                    # Title slide should have only Title pleceholder type, etc.
                    foreach (ILayoutSlide titleAndObjectLayoutSlide in layoutSlides)
                    {
                        if (titleAndObjectLayoutSlide.Name == "Title and Object")
                        {
                            layoutSlide = titleAndObjectLayoutSlide
                            break
                        }
                    }

                    if (layoutSlide == None)
                    {
                        foreach (ILayoutSlide titleLayoutSlide in layoutSlides)
                        {
                            if (titleLayoutSlide.Name == "Title")
                            {
                                layoutSlide = titleLayoutSlide
                                break
                            }
                        }

                        if (layoutSlide == None)
                        {
                            layoutSlide = layoutSlides.GetByType(SlideLayoutType.Blank)
                            if (layoutSlide == None)
                            {
                                layoutSlide = layoutSlides.add(SlideLayoutType.TitleAndObject, "Title and Object")
                            }
                        }
                    }
                }

                # Adding empty slide with added layout slide 
                presentation.slides.InsertEmptySlide(0, layoutSlide)

                # Save presentation    
                presentation.save(dataDir + "AddLayoutSlides_out.pptx", slides.export.SaveFormat.PPTX)
            }
            #ExEnd:AddLayoutSlides
        }
    }
}

File: examples/src/Slides/layout/add_layout_slides.py
import aspose.slides as slides


def add_layout_slides(global_opts):
    # Instantiate Presentation class that represents the presentation file
    with slides.Presentation(global_opts.data_dir + "welcome-to-powerpoint.pptx") as presentation:
        # Try to search by layout slide type
        layout_slides = presentation.masters[0].layout_slides
        layout_slide = layout_slides.get_by_type(slides.SlideLayoutType.TITLE_AND_OBJECT) and layout_slides.get_by_type(
            slides.SlideLayoutType.TITLE)

        if layout_slide is None:
            # The situation when a presentation doesn't contain some type of layouts.
            # presentation File only contains Blank and Custom layout types.
            # But layout slides with Custom types has different slide names,
            # like "Title", "Title and Content", etc. And it is possible to use these
            # names for layout slide selection.
            # Also it is possible to use the set of placeholder shape types. For example,
            # Title slide should have only Title pleceholder type, etc.
            for title_and_object_layout_slide in layout_slides:
                if title_and_object_layout_slide.name == "Title and Object":
                    layout_slide = title_and_object_layout_slide
                    break
            if layout_slide is None:
                for titleLayoutSlide in layout_slides:
                    if titleLayoutSlide.name == "Title":
                        layout_slide = titleLayoutSlide
                        break

                if layout_slide is None:
                    layout_slide = layout_slides.get_by_type(slides.SlideLayoutType.BLANK) or layout_slides.add(slides.SlideLayoutType.TITLE_AND_OBJECT, "Title and Object")

        # Adding empty slide with added layout slide
        presentation.slides.insert_empty_slide(0, layout_slide)

        # Save presentation
        presentation.save(global_opts.out_dir + "layout_add_layout_slides_out.pptx", slides.export.SaveFormat.PPTX)
